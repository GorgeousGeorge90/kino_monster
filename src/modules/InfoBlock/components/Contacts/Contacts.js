import styles from './Contacts.module.scss';


const Contacts = () => {

    return (<section className={styles.container}>
            <h3>Contacts</h3>
            <ul>
                <li>
                    <span>Email:</span>
                    <span>Gambit47@yandex.ru</span>
                </li>
                <li>
                    <span>GitHub:</span>
                    <span><a href='https://github.com/GorgeousGeorge90'>https://github.com/GorgeousGeorge90</a></span>
                </li>
            </ul>
        </section>)
}

export default Contacts