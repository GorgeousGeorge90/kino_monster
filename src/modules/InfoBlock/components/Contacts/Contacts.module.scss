

.container {
  @apply flex flex-col items-center gap-2;

  h2 {
    @apply font-bold text-lg text-neutral-100 font-mono;
  }

  li {
    @apply flex gap-2;

    .contacts_info {
      @apply  font-bold text-neutral-300 font-mono;
    }

    .contacts_data {
      @apply text-neutral-400 font-serif;
    }

    a {
      @apply underline text-blue-500 decoration-blue-500 dark:text-blue-700 dark:decoration-blue-700;
    }

  }
}