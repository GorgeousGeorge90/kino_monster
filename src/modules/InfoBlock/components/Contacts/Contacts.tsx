import styles from './Contacts.module.scss';


const Contacts = () => {

    return (<section className={styles.container}>
            <h2>Contacts</h2>
            <ul>
                <li>
                    <span className={styles.contacts_info}>Email:</span>
                    <span className={styles.contacts_data}>Gambit47@yandex.ru</span>
                </li>
                <li>
                    <span className={styles.contacts_info}>GitHub:</span>
                    <span className={styles.contacts_data}><a href='https://github.com/GorgeousGeorge90'>https://github.com/GorgeousGeorge90</a></span>
                </li>
            </ul>
        </section>)
}

export default Contacts