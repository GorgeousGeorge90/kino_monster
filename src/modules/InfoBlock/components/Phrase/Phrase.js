import styles from './Phrase.module.scss';

const Phrase = ({quote,author}) => {

    return (<section className={styles.container}>
            <h2>{quote}</h2>
            <p>{author}</p>
        </section>)
}

export default Phrase