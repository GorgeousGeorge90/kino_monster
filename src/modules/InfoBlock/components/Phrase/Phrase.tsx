import styles from './Phrase.module.scss';
import { FC } from 'react';
import { PhraseItemType } from '../../types/types';
import {useAppSelector} from "../../../../store/hooks";
import {getPhrase} from "../../selectors/selectors";


const Phrase = () => {
    const phrase = useAppSelector(getPhrase)

    return (<section className={styles.phrase_container}>
            <h2 className={styles.phrase_title}>{phrase?.quote}</h2>
            <p className={styles.phrase_author}>{phrase?.author}</p>
    </section>)
}

export default Phrase