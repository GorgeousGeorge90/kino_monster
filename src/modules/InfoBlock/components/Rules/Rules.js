import styles from './Rules.module.scss';


const Rules = () => {

    return (<section className={styles.container}>
        <h2>Rules</h2>
        <ul className={styles.rules_list}>
            <li>1.don't be rude in comments!</li>
            <li>2.fill films inputs with correct data!</li>
            <li>3.comments limit is 5!</li>
            <li>3.have a nice day!</li>
        </ul>
    </section>)
}

export default Rules