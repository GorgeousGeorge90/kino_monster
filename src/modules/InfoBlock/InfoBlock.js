import styles from './InfoBlock.module.scss';
import {useDispatch, useSelector} from "react-redux";
import {fetchPhrase} from "./store/infoSlice";
import Rules from "./components/Rules/Rules";
import Contacts from "./components/Contacts/Contacts";
import Phrase from "./components/Phrase/Phrase";
import {useEffect} from "react";
import {getPhrase} from "./selectors/selectors";


const InfoBlock = () => {
    const dispatch = useDispatch()
    useEffect(()=> {
        dispatch(fetchPhrase())
    },[dispatch])
    const phrase = useSelector(state=>getPhrase(state))

    return (<div className={styles.container}>
        <Rules/>
        <Contacts/>
        <Phrase {...phrase}/>
    </div>)
}

export default InfoBlock