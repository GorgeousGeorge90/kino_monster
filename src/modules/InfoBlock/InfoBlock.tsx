import styles from './InfoBlock.module.scss';
import { fetchPhrase } from './store/infoSlice';
import Rules from "./components/Rules/Rules";
import Phrase from "./components/Phrase/Phrase";
import { useEffect } from 'react';
import { getLoading, getPhrase } from './selectors/selectors';
import { ReactComponent as Spinner } from '../../assets/img/spinner.svg';
import { useAppDispatch, useAppSelector}  from '../../store/hooks';


const InfoBlock = () => {
    const dispatch = useAppDispatch()
    useEffect(()=> {
        dispatch(fetchPhrase())
    },[dispatch])

    const status = useAppSelector(getLoading)

    return (<div className={styles.info_container}>
        <img className={styles.info_pic}
            src="https://media.istockphoto.com/id/489833698/photo/two-lounge-chairs-under-tent-on-beach.jpg?s=612x612&w=0&k=20&c=df_6Jkb5YreyaAcGVbdcRrTP4dZ-opUrYWtgRC3eKKI=" alt='img'/>
        <Rules/>
        {
            status === 'pending' ? <Spinner/>:<Phrase/>
        }
    </div>)
}

export default InfoBlock