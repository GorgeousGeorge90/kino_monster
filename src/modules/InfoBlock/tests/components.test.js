import {render,screen} from "@testing-library/react";
import Contacts from "../components/Contacts/Contacts";
import Phrase from "../components/Phrase/Phrase";
import Rules from "../components/Rules/Rules";
import * as reduxHooks from "react-redux";
import InfoBlock from "../InfoBlock";


jest.mock('react-redux')
const mockedDispatch = jest.spyOn(reduxHooks, 'useDispatch')
const mockedSelector = jest.spyOn(reduxHooks, 'useSelector')


describe('render tests for InfoBlock',()=> {
    it('goal: render Contacts component',()=> {
        render(<Contacts/>)

        const component = screen.getByRole('heading')
        expect(component).toBeInTheDocument()
        expect(component).toHaveTextContent('Contacts')
    })

    it('goal: render Phrase component',()=> {
        const data = {
            quote: 'I am rich!',
            author: 'Tom Ford'
        }

        render(<Phrase {...data}/>)
        const component = screen.getByRole('heading')
        expect(component).toBeInTheDocument()
        expect(component).toHaveTextContent('I am rich!')
    })

    it('goal: render Rules component',()=> {

        render(<Rules/>)
        const component = screen.getByRole('heading')
        expect(component).toBeInTheDocument()
        expect(component).toHaveTextContent('Rules')
    })

    it('goal: render InfoBlock component',()=> {
        const data = {
            phrase: {
                quote: 'I am rich!',
                author: 'Tom Ford'
            },
            status:'resolved',
        }

        const dispatch = jest.fn()

        mockedSelector.mockReturnValue(data)
        mockedDispatch.mockReturnValue(dispatch)
        const view = render(<InfoBlock/>)
        expect(view).toMatchSnapshot()
    })
})