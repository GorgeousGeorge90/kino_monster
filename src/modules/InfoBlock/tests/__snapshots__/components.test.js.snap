// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render tests for InfoBlock goal: render InfoBlock component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="info_container"
      >
        <img
          alt="img"
          class="info_pic"
          src="https://media.istockphoto.com/id/489833698/photo/two-lounge-chairs-under-tent-on-beach.jpg?s=612x612&w=0&k=20&c=df_6Jkb5YreyaAcGVbdcRrTP4dZ-opUrYWtgRC3eKKI="
        />
        <section
          class="rules_container"
        >
          <h2
            class="rules_title"
          >
            Rules
          </h2>
          <ul
            class="rules_list"
          >
            <li>
              1.don't be rude in comments!
            </li>
            <li>
              2.fill films inputs with correct data!
            </li>
            <li>
              3.comments limit is 5!
            </li>
            <li>
              3.have a nice day!
            </li>
          </ul>
        </section>
        <section
          class="phrase_container"
        >
          <h2
            class="phrase_title"
          />
          <p
            class="phrase_author"
          />
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="info_container"
    >
      <img
        alt="img"
        class="info_pic"
        src="https://media.istockphoto.com/id/489833698/photo/two-lounge-chairs-under-tent-on-beach.jpg?s=612x612&w=0&k=20&c=df_6Jkb5YreyaAcGVbdcRrTP4dZ-opUrYWtgRC3eKKI="
      />
      <section
        class="rules_container"
      >
        <h2
          class="rules_title"
        >
          Rules
        </h2>
        <ul
          class="rules_list"
        >
          <li>
            1.don't be rude in comments!
          </li>
          <li>
            2.fill films inputs with correct data!
          </li>
          <li>
            3.comments limit is 5!
          </li>
          <li>
            3.have a nice day!
          </li>
        </ul>
      </section>
      <section
        class="phrase_container"
      >
        <h2
          class="phrase_title"
        />
        <p
          class="phrase_author"
        />
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
